src/date/format-properties.js

```diff
@@ -103,19 +109,25 @@ return function( pattern, cldr ) {
 			// Month
 			case "M":
 			case "L":
+				properties.months = properties.months || {};
+				properties.months[ chr ] = properties.months[ chr ] || {};
 				if ( length > 2 ) {
-					if ( !properties.months ) {
-						properties.months = {};
-					}
-					if ( !properties.months[ chr ] ) {
-						properties.months[ chr ] = {};
-					}
 					properties.months[ chr ][ length ] = cldr.main([
-						"dates/calendars/gregorian/months",
+						"dates/calendars",
+						properties.calendar,
+						"months",
 						chr === "M" ? "format" : "stand-alone",
 						widths[ length - 3 ]
 					]);
-				} else {
+				}
+				properties.months[ chr ][ length ] = dateMonthNames(
+					properties.months[ chr ][ length ],
+					chr,
+					length,
+					properties.calendar,
+					cldr
+				);
+				if ( length <= 2 ) {
 					formatNumber = true;
 				}
```

Comment:
Unfortunately, the CLDR seems to specify that Chinese leap months get names like "6bis" or "06bis" even for "numeric" months. So properties.months is still needed. 
The code initially sets it from CLDR, then uses dateMonthNames to update the names of the leap months.

I can move that final `if` back into the else.